src/JWT/Builder/JwtBuilder.cs

```diff
@@ -192,8 +198,15 @@ public JwtBuilder WithSecret(params byte[][] secrets)
         /// Instructs to verify the JWT signature.
         /// </summary>
         /// <returns>Current builder instance</returns>
-        public JwtBuilder MustVerifySignature() =>
-            WithVerifySignature(true);
+        public JwtBuilder MustVerifySignature()
+        {
+            if (String.Equals(algorithm?.Name, nameof(AlgorithmName.None), StringComparison.OrdinalIgnoreCase))

```

Comment:
Please move this check to `WithVerifySignature(bool)`

Reply:
done in latest push